src/drivers/power_monitor/ina228/ina228.cpp

```diff
@@ -124,7 +124,12 @@ int INA228::read(uint8_t address, int32_t &data)
 	const int ret = transfer(&address, 1, (uint8_t *)&received_bytes, sizeof(received_bytes) - 1);
 
 	if (ret == PX4_OK) {
-		data = swap32(received_bytes) >> ((32 - 24) + 4);
+		data = swap32(received_bytes) >> ((32 - 24) + 4); // Convert to 20bit value
```

Comment:
One comment says 24 bit, another comment says 20 bit?

Anyway would it be simpler and overall clearer to just deal with the 3 bytes directly? You could do the same thing for different sizes.

 - transfer 3 bytes
 - explicitly arrange those bytes into a signed 32 bit integer
 - shift by 8 to form 24 bit signed integer in an int32


Something like this?
``` C
uint8_t received_bytes[3]{};
const int ret = transfer(&address, 1, received_bytes, sizeof(received_bytes));

int32_t value = ((received_bytes[0] << 24) | (received_bytes[1] << 16) | (received_bytes[2] << 8)) >> 8;  

```

Reply:
I agree completely. I started to rewrite the driver but I just don't have the time right now to redo it all and validate it so I figured I'd push the fix that does actually fix a bug, while the rest of the refactor would be making things more clear/correct